src/Database/Type/DateTimeType.php

```diff
@@ -53,9 +53,12 @@ class DateTimeType extends Type implements TypeInterface
     /**
      * String format to use for DateTime parsing
      *
-     * @var string
+     * @var string|array
      */
-    protected $_format = 'Y-m-d H:i:s';
+    protected $_format = [
+        'Y-m-d H:i:s',
+        'Y-m-d\TH:i:s.uO',
```

Comment:
Actually i ran into this issue because we started testing with MySQL 5.7, and the same app/db on MySQL 5.6 this "just works" (posting json from a js frontend to a controller)